middleware/verifyUser.js

```diff
@@ -0,0 +1,53 @@
+import { compareSync } from 'bcrypt';
+import { User } from '../models';
+
+let userWantsToLogin = false;
+
+export default (req, res, next) => {
+  const {
+    body,
+    user,
+  } = req;
+  const options = {};
+  if (user) {
+    options.fieldName = 'id';
+    options.fieldValue = user.id;
+  } else if (body) {
+    options.fieldName = 'email';
+    options.fieldValue = body.email;
+    userWantsToLogin = true;
+  }
+
+  const { fieldName, fieldValue } = options;
+  User.findOne({ where: { [fieldName]: fieldValue } })
+    .then((foundUser) => {
+      if (foundUser) {
+        const { dataValues: { verified, password } } = foundUser;
+        if (!verified) {
+          return res.status(403).json({
+            success: false,
+            errors: [
```

Comment:
The three lines can be compressed into one

Reply:
They are not three lines per se, its just the formatting that makes it look like that. It's easier to read in this format.